src/transport/Session.h

```diff
@@ -78,6 +78,10 @@ class Session
     virtual System::Clock::Timestamp GetMRPBaseTimeout()               = 0;
     virtual System::Clock::Milliseconds32 GetAckTimeout() const        = 0;
 
+    // Returns a suggested timeout value based on the session type and the given upper layer processing timeout.
+    // For group sessions, this function will always return 0.
+    System::Clock::Timeout SuggestEndToEndTimeout(System::Clock::Timeout upperlayerProcessingTimeout);
```

Comment:
I think the term 'end to end' is somewhat vague here.

e.g `ComputeRoundTripTimeout`?